### DISCUSSION TITLE
自定义顶栏搜索没有联想

### DISCUSSION THREAD

**Discussion by ChenJW431** [the1812/Bilibili-Evolved#4577] [提问 / Q&A]
![屏幕截图 2023-12-21 121604](https://github.com/the1812/Bilibili-Evolved/assets/122283633/5487688a-5fdb-4905-8b55-817dc083fee0)

**Comment by the1812** [comment]
这是搜索联想的 API, 试试看能打开不?
https://s.search.bilibili.com/main/suggest?func=suggest&suggest_type=accurate&sub_type=tag&main_ver=v1&highlight=&userid=&bangumi_acc_num=1&special_acc_num=1&topic_acc_num=1&upuser_acc_num=3&tag_num=10&special_num=10&bangumi_num=10&upuser_num=3&term=%E6%88%91

**Reply by ChenJW431** [reply]
打开只显示这些{"exp_str":"","code":0,"result":{"tag":[{"value":"æˆ‘æ˜¯æ¡¶å­å“¥","term":"æˆ‘æ˜¯æ¡¶å­å“¥","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>æ˜¯æ¡¶å­å“¥","spid":5,"type":""},{"value":"æˆ‘çš„ä¸–ç•Œ","term":"æˆ‘çš„ä¸–ç•Œ","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>çš„ä¸–ç•Œ","spid":5,"type":""},{"value":"æˆ‘ä¸åƒç‰›è‚‰","term":"æˆ‘ä¸åƒç‰›è‚‰","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>ä¸åƒç‰›è‚‰","spid":5,"type":""},{"value":"æˆ‘æœŸå¾…çš„ä¸æ˜¯é›ª","term":"æˆ‘æœŸå¾…çš„ä¸æ˜¯é›ª","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>æœŸå¾…çš„ä¸æ˜¯é›ª","spid":5,"type":""},{"value":"æˆ‘ä¸åƒç‰›è‚‰æ˜¯ä»€ä¹ˆæ¢—","term":"æˆ‘ä¸åƒç‰›è‚‰æ˜¯ä»€ä¹ˆæ¢—","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>ä¸åƒç‰›è‚‰æ˜¯ä»€ä¹ˆæ¢—","spid":5,"type":""},{"value":"æˆ‘æŠ«æ˜Ÿæˆ´æœˆå¥‹ä¸é¡¾èº«çš„äººæ€Žä¹ˆæ ·äº†äºŒæ‰‹è½¦è´©å­","term":"æˆ‘æŠ«æ˜Ÿæˆ´æœˆå¥‹ä¸é¡¾èº«çš„äººæ€Žä¹ˆæ ·äº†äºŒæ‰‹è½¦è´©å­","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>æŠ«æ˜Ÿæˆ´æœˆå¥‹ä¸é¡¾èº«çš„äººæ€Žä¹ˆæ ·äº†äºŒæ‰‹è½¦è´©å­","spid":5,"type":""},{"value":"æˆ‘è¿˜èƒ½è¯´ä»€ä¹ˆå‘¢æ›¼å·´è½å¹•","term":"æˆ‘è¿˜èƒ½è¯´ä»€ä¹ˆå‘¢æ›¼å·´è½å¹•","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>è¿˜èƒ½è¯´ä»€ä¹ˆå‘¢æ›¼å·´è½å¹•","spid":5,"type":""},{"value":"æˆ‘æ˜¯ç‘žæ–¯æ‹œ","term":"æˆ‘æ˜¯ç‘žæ–¯æ‹œ","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>æ˜¯ç‘žæ–¯æ‹œ","spid":5,"type":""},{"value":"æˆ‘çš„ä¸–ç•Œçå¦®æ¨¡ç»„javaæ‰‹æœºç‰ˆå¦‚ä½•ä¸‹è½½","term":"æˆ‘çš„ä¸–ç•Œçå¦®æ¨¡ç»„javaæ‰‹æœºç‰ˆå¦‚ä½•ä¸‹è½½","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>çš„ä¸–ç•Œçå¦®æ¨¡ç»„javaæ‰‹æœºç‰ˆå¦‚ä½•ä¸‹è½½","spid":5,"type":""},{"value":"æˆ‘æƒ³å¿µ","term":"æˆ‘æƒ³å¿µ","ref":0,"name":"<em class=\"suggest_high_light\">æˆ‘</em>æƒ³å¿µ","spid":5,"type":""}]},"stoken":"2367434809481204459"}

**Reply by the1812** [reply]
你这浏览器编码不太对吧... 
正确的应该是这样的:
<code>{"exp_str":"","code":0,"result":{"tag":[{"value":"我想念","term":"我想念","ref":0,"name":"<em class=\"suggest_high_light\">我</em>想念","spid":5,"type":""},{"value":"我的世界","term":"我的世界","ref":0,"name":"<em class=\"suggest_high_light\">我</em>的世界","spid":5,"type":""},{"value":"我是桶子哥","term":"我是桶子哥","ref":0,"name":"<em class=\"suggest_high_light\">我</em>是桶子哥","spid":5,"type":""},{"value":"我不吃牛肉","term":"我不吃牛肉","ref":0,"name":"<em class=\"suggest_high_light\">我</em>不吃牛肉","spid":5,"type":""},{"value":"我期待的不是雪","term":"我期待的不是雪","ref":0,"name":"<em class=\"suggest_high_light\">我</em>期待的不是雪","spid":5,"type":""},{"value":"我不吃牛肉是什么梗","term":"我不吃牛肉是什么梗","ref":0,"name":"<em class=\"suggest_high_light\">我</em>不吃牛肉是什么梗","spid":5,"type":""},{"value":"我披星戴月奋不顾身的人怎么样了二手车贩子","term":"我披星戴月奋不顾身的人怎么样了二手车贩子","ref":0,"name":"<em class=\"suggest_high_light\">我</em>披星戴月奋不顾身的人怎么样了二手车贩子","spid":5,"type":""},{"value":"我还能说什么呢曼巴落幕","term":"我还能说什么呢曼巴落幕","ref":0,"name":"<em class=\"suggest_high_light\">我</em>还能说什么呢曼巴落幕","spid":5,"type":""},{"value":"我是瑞斯拜","term":"我是瑞斯拜","ref":0,"name":"<em class=\"suggest_high_light\">我</em>是瑞斯拜","spid":5,"type":""},{"value":"我的世界珍妮模组java手机版如何下载","term":"我的世界珍妮模组java手机版如何下载","ref":0,"name":"<em class=\"suggest_high_light\">我</em>的世界珍妮模组java手机版如何下载","spid":5,"type":""}]},"stoken":"4769669783378058540"}</code>

**Reply by ChenJW431** [reply]
我上网查了半天，firefox没有改文字编码的选项，只有按钮能修复网页的文字编码，不知道用什么办法修复搜索框的文字编码。我不懂这些，所以如果有精力的话能否做一个强制改变文字编码的脚本或者插件？
